### ISSUE TITLE
I heard you lacked motivation!? 

### ISSUE THREAD

**Issue by HamzaMateen** [kardwen/passepartui#8]
please keep developing and refining this project if its not yet finished. That ui is just beautiful. I am huge fan of pass and use exclusively android client password store or the cli itself to manage my secrets but with that ui, I can finally have some real peace. 

Thank you for your efforts!

**Comment by kardwen** [comment]
Thank you, I will continue to work on the project, but depending on what else is going on, it may take a while.

**Comment by grouzen** [comment]
I'm also a massive fan of pass and I'm using it in my terminal and on Android. Passepartui doesn't work well in my case (urxvt terminal on Linux). The interface falls apart when I want to copy a password. I think it may be related to the terminal I use. So I'm thinking about allocating some of my time to fix at least this issue as I think this project is awesome.

**Comment by kardwen** [comment]
That would be great! I created an issue (https://github.com/kardwen/passepartui/issues/9) to bundle everything related to this.

I think you may be able to copy a password if you authenticate before starting passepartui.

**Comment by grouzen** [comment]
> I think you may be able to copy a password if you authenticate before starting passepartui.

Checked this, you're right. I don't know if that's how you initially designed this but I personally would like to be able to enter a passphrase whenever gpg asks me, including when I try to copy the password from within the passepartui

**Comment by grouzen** [comment]
@kardwen I spent some time researching this, and I think the most viable option is to run pass/gpg as a first step during app initialization to force a user to enter a passphrase so that it is cached by gpg-agent for the rest of the session (there are some nuances around cache TTL). I still need to understand how to make gpg/pass to ask for the passphrase without giving it a specific file/data for decoding. So, I wanted to ask you about your current authenticating flow before you start the app.
I also was tinkering with the idea of forcing pass/gpg to read the passphrase from the input so we can show our own custom dialog within the app. For example, you can do it by passing additional parameters to gpg via `PASSWORD_STORE_GPG_OPTS` env variable like:
```
PASSWORD_STORE_GPG_OPTS="--pinentry-mode loopback --passphrase-fd 0" pass <your/password/file>
```
But it seems that in this case it waits for input regardless of whether you entered the password a second ago or not.

And I don't think it is possible to show a standard or custom pinentry dialog because it implies running another CLI app (it what happens now when I copy the password not being authenticated). It wipes out the UI rendered by ratatui and we need somehow to re-initialize it again after the pinenetry dialog is closed.

**Comment by grouzen** [comment]
Ok, I just figured out how to easily get gpg-agent to cache the passphrase. To do this, we can run `gpg --sign -a` to create a phone signed message so that gpg can call gpg-agent, which will ask for the passphrase to cache.

**Comment by kardwen** [comment]
I think the easiest way is to add a configuration file that allows you to set an option to ask for the password on startup (if the password is not cached). I plan to add a configuration file later anyway, for example for a color theme.

I have not yet thought about the implementation of the password query.

**Comment by grouzen** [comment]
Sure, configuration is helpful in this case. I'm more curious about how other people use the app. 
Can you describe your flow?

I can imagine the following:
1. call `pass some/service/password`. It makes the gpg-agent show the pinentry dialog.
2. enter the password 
3. run the app and copy whatever passwords you want

**Comment by kardwen** [comment]
passepartui currently calls pass directly, so any pass operation that would normally require authentication also requires authentication when run by passepartui, for example `pass <some-password> show`  or `pass <some-password> --clip`.

Implemented here (serial function that is then run in a new thread), it is really simple:
https://github.com/kardwen/passepartout/blob/5766f78a064ad62911357a51d21f425aee21093c/src/password_store.rs#L41-L55

The pinentry program is executed and captures all inputs (some information can be found here: https://wiki.archlinux.org/title/GnuPG). Naturally it closes again when the correct password is entered or the process is aborted.

**Comment by grouzen** [comment]
Yeah, I know all these details. I was asking about the way you use the app.

**Comment by kardwen** [comment]
Sorry for misunderstanding your question. I hope this helps:

When designing the user interaction, I wanted to focus on quick and easy exploration of the password store. The navigation in the TUI and the filter function are designed with this in mind.

I sometimes copy passwords with the TUI, but often I simply use pass, for example in an e-mail program such as aerc or in a script for a VPN connection. I currently use passff for browser passwords in Firefox.

I also think that the function for copying the pass ID might be useful to run a pass command in a separate terminal window for example `pass edit`. It would be better of course If passepartui would register updates for the password list, which is not yet implemented,

**Comment by grouzen** [comment]
Thanks! I'm asking because I'm trying to understand if I should put an effort into implementing this functionality. Now I see that your flow and the original idea behind this were different, which is fine.

**Comment by kardwen** [comment]
What do you have in mind? Maybe it could work anyways.

**Comment by kardwen** [comment]
I also want to add new functions for editing or generating passwords but this is more difficult. And I definitely need to write some tests and have other people audit my code for more advanced features. With the current implementation there simply is a lot less that can go wrong.

**Comment by grouzen** [comment]
> What do you have in mind? Maybe it could work anyways.

I want to be able to copy passwords from within the app

**Comment by kardwen** [comment]
I just installed `rxvt-unicode` on Debian and had no issues copying stuff to my clipboard with passepartui. I'm not sure I'm understanding correctly what kind of setup you have.

**Comment by grouzen** [comment]
@kardwen In my case UI breaks apart when I'm trying to copy the password not being logged in into pass/gpg. It works fine when I'm logged in before running passepartui. I don't think using other terminal emulators makes much difference.

**Comment by kardwen** [comment]
Closing this in favour of https://github.com/kardwen/passepartui/issues/9
